File: george/src/test/java/ui/pages/SearchPage.java
package ui.pages;

import com.codeborne.selenide.Selenide;
import com.codeborne.selenide.SelenideElement;
import org.openqa.selenium.By;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import ui.BaseTest;

import static com.codeborne.selenide.Selenide.$;
import static com.codeborne.selenide.Selenide.$x;

public class SearchPage extends BaseTest {
    public final SelenideElement linkSearch = $x("//a[contains(@href,'/search')]");
    public final SelenideElement textField = $x("//input");
    public SelenideElement searchButton = $x("//button");
    public final SelenideElement resultSearch = $(By.xpath("//div[contains(@class,'alert')]"));

    @FindBy(xpath = "//input")
    private WebElement textFieldInput;

    public void searchFor(String text) {
        textField.setValue(text).pressEnter();
    }

    public void openWebsite(String url) {
        Selenide.open(url);
    }

    public void clickLinkSearch() {
        linkSearch.click();
    }
}

File: george/src/test/kotlin/com/georgeCross/tests/pages/SearchPage.kt
package com.georgeCross.tests.pages

import com.codeborne.selenide.Selenide.*

class SearchPage {

    private val inputNumber = `$x`("//input")
    private val searchButton = `$x`("//button[contains(text(), 'Искать')]")
    private val badge = `$x`("//span[contains(., '231440')]")

    fun searchNumber(){
        inputNumber.click()
        inputNumber.sendKeys("231440")
        searchButton.click()
        badge.isDisplayed
    }
}